### PR TITLE
Add integration tests for DotNet

### PR DESCRIPTION
**Pull request by Paramadon**
*Issue #, if available:*
This pr add to add integration test to dot net adds default instrumentation tests and auto annotation tests for dotnet.

Here is the passing test runs for dot net instrumentation test:
Screenshot 2024-07-12 at 12 02 25 PM
Screenshot 2024-07-12 at 12 02 03 PM
Screenshot 2024-07-12 at 12 01 21 PM
Screenshot 2024-07-12 at 11 56 57 AM

*Description of changes:*
 Added defaul instrumentation test in the operator integration workflow

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by musa-asad — APPROVED** [review]

File: .github/workflows/operator-integration-test.yml [hunk 1]
```diff
@@ -55,27 +55,14 @@ jobs:
           make deploy
           kubectl wait --for=condition=Ready pod --all -n amazon-cloudwatch
 
-      - name: Test case for cloudwatch agent pod creation
```
Comment (lisguo) [inline .github/workflows/operator-integration-test.yml:58]: why was this removed?

File: .github/workflows/operator-integration-test.yml [hunk 2]
```diff
@@ -104,31 +90,51 @@ jobs:
...
-          kubectl wait --for=condition=Ready pod --all -n default
-          
-          go run integration-tests/manifests/cmd/validate_instrumentation_vars.go default integration-tests/python-java/default_instrumentation_python-java_env_variables.json
+          kubectl wait --for=condition=Ready pod --all -n default  
+          cd integration-tests/manifests/cmd
+          go run validate_instrumentation_vars.go default ns_instrumentation_env_variables.json
+          kubectl delete instrumentation sample-instrumentation
+
+
+      - name: Test for default instrumentation resources for python java dotnet
```
Comment (lisguo) [inline .github/workflows/operator-integration-test.yml:118]: imo this should just be `Test for default instrumentation resources for all languages` and then you don't have to pile on each language. Same for the directory name `integration-tests/python-java-dot-net` -> `integration-tests/all-languages`

File: integration-tests/dotnet/default_instrumentation_dotnet_env_variables.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "OTEL_TRACES_SAMPLER_ARG": "endpoint=http://cloudwatch-agent.amazon-cloudwatch:2000",
+  "OTEL_TRACES_SAMPLER" : "xray",
+  "OTEL_EXPORTER_OTLP_PROTOCOL": "http/protobuf",
+  "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT": "http://cloudwatch-agent.amazon-cloudwatch:4316/v1/traces",
+  "OTEL_METRICS_EXPORTER": "none",
+  "OTEL_DOTNET_DISTRO": "aws_distro",
+  "OTEL_LOGS_EXPORTER": "none"
```
Comment (lisguo) [inline integration-tests/dotnet/default_instrumentation_dotnet_env_variables.json:8]: we want to include more variables than these for dotnet. We want everything here: https://github.com/aws/amazon-cloudwatch-agent-operator/blob/main/pkg/instrumentation/defaultinstrumentation_test.go#L72

There might be more as well such as 
```
CORECLR_PROFILER_PATH:                           /otel-auto-instrumentation-dotnet/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so
      DOTNET_STARTUP_HOOKS:                            /otel-auto-instrumentation-dotnet/net/OpenTelemetry.AutoInstrumentation.StartupHook.dll
      DOTNET_ADDITIONAL_DEPS:                          /otel-auto-instrumentation-dotnet/AdditionalDeps
      OTEL_DOTNET_AUTO_HOME:                           /otel-auto-instrumentation-dotnet
      DOTNET_SHARED_STORE:                             /otel-auto-instrumentation-dotnet/store
```

But I'd probably cross reference with what you are currently seeing in your pod to ensure these are accurate

File: .github/workflows/operator-integration-test.yml (hunk 1 above)
Comment (Paramadon) [reply .github/workflows/operator-integration-test.yml:58]: This test felt repetitve because the other part of the integ tests pod creation, but I can see why this might be useful and will add it back.

File: .github/workflows/operator-integration-test.yml (hunk 2 above)
Comment (Paramadon) [reply .github/workflows/operator-integration-test.yml:118]: Ahh yes this will be useful when we add more lanugages, made the change.

File: integration-tests/dotnet/default_instrumentation_dotnet_env_variables.json (hunk 3 above)
Comment (Paramadon) [reply integration-tests/dotnet/default_instrumentation_dotnet_env_variables.json:8]: Oh wait yes your right.  "CORECLR_ENABLE_PROFILING": "1" is also missing it seems. Just added all of these

File: integration-tests/manifests/annotations/validate_annotations_deployment_test.go [hunk 4]
```diff
@@ -127,3 +133,37 @@ func TestPythonOnlyDeployment(t *testing.T) {
...
 }
+func TestDotNetOnlyDeployment(t *testing.T) {
+
+	clientSet := setupTest(t)
+	randomNumber, err := rand.Int(rand.Reader, big.NewInt(9000))
+	if err != nil {
+		panic(err)
+	}
+	randomNumber.Add(randomNumber, big.NewInt(1000)) //adding a hash to namespace
+	uniqueNamespace := fmt.Sprintf("deployment-namespace-python-only-%d", randomNumber)
```
[the remark this answers is not in the document]
Comment (Paramadon) [reply integration-tests/manifests/annotations/validate_annotations_deployment_test.go]: changed to dotnet for all

**Review by lisguo — APPROVED** [review]

**Review by musa-asad — APPROVED** [review]
